app/src/unifiedpush/java/im/molly/unifiedpush/UnifiedPushDistributor.kt

```diff
@@ -19,26 +18,34 @@ object UnifiedPushDistributor {
     UnifiedPush.unregisterApp(AppDependencies.application)
   }
 
-  fun selectFirstDistributor() {
+  fun selectCurrentOrDefaultDistributor() {
     val context = AppDependencies.application
-    UnifiedPush.getSavedDistributor(context)
-      ?: UnifiedPush.getDistributors(context).firstOrNull()?.also {
-        UnifiedPush.saveDistributor(context, it)
+    UnifiedPush.tryUseCurrentOrDefaultDistributor(context) { success ->
```

Comment:
It doesn't work because the context isn't an activity, it is resolved here: https://github.com/mollyim/mollyim-android/pull/390/commits/48bd466c8750f18de54edcb756cd2b1c2599a79a